### ISSUE TITLE
Limited forward compatibility via using grammar for previous version

### ISSUE THREAD

**Issue by Kojoley** [davidhalter/parso#201]
The second alpha CPython 3.11 release had been out for a while https://www.python.org/download/pre-releases/. In IPython I had to disable quite a bunch of tests because parso has no grammar for Python 3.11. I don't know if there some new grammar features in Python 3.11 or the grammar is not stable yet and parso is not rushing implementing it, but IPython uses only constructs presented before Python 3.8 which could be possible to parse on any interpreter with parso that implements only 3.8 grammar.

Is it possible to make jedi fallback gracefully to a grammar for previous Python version instead of propagating `NotImplementedError` exception from parso?

**Comment by Kojoley** [comment]
Oh, and I opened the issue on parso repository, I am sorry, could you please transfer it to jedi?

**Comment by isidentical** [comment]
I think `parso` is the right place. We can add a grammar for 3.11, though we do not even properly support 3.10 so I am not really sure if that is the right way to go. CC: @davidhalter

**Comment by davidhalter** [comment]
Adding the 3.11 grammar is the right step forward. Proper support for 3.10/3.11 is a different issue, but I guess the IPython guys will not really be affected by this anyway. Autocompletion should work even without proper understanding of match syntax (which most people won't even use in a REPL.

**Comment by isidentical** [comment]
Seems like we already have a grammar for 3.11;
```py
>>> parso.load_grammar(version='3.11')
<PythonGrammar:single_input file_input eval_input ...>
```

https://github.com/davidhalter/parso/commit/f2b1ff942933a43c34d93e3359e7115d20cc2ee3

**Comment by Kojoley** [comment]
> Seems like we already have a grammar for 3.11;
> 
> ```python
> >>> parso.load_grammar(version='3.11')
> <PythonGrammar:single_input file_input eval_input ...>
> ```
> 
> [f2b1ff9](https://github.com/davidhalter/parso/commit/f2b1ff942933a43c34d93e3359e7115d20cc2ee3)

Did it miss some steps to be included in the wheel? I don't see the file inside the https://files.pythonhosted.org/packages/a9/c4/d5476373088c120ffed82f34c74b266ccae31a68d665b837354d4d8dc8be/parso-0.8.2-py2.py3-none-any.whl

![image](https://user-images.githubusercontent.com/2743474/143690358-2e93f5dd-0f11-4d98-a259-a650e3da79d1.png)

**Comment by Kojoley** [comment]
Well, the change was simply not released yet, I did not sport that the commit is in 'May' and the release was in 'Mar'.

**Comment by davidhalter** [comment]
Released 0.8.3, because that might be useful to other core devs as well. Also added a 3.12 grammar, because people are probably going to ask for that in a year from now.
